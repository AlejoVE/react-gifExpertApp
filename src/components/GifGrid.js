import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<>
			<h3 className="animate__animated animate__fadeIn">{category}</h3>
			<div className="card-grid">
				{/* este simbolo && significa que si está cargando, ponga un párrago que diga loading
				Si no está cargando, no hace nada */}
				{loading && <p className="animate__animated animate__flash">Loading...</p>}
				{
					// prettier/ignore
					images.map((img) => (
						<GifGridItem {...img} key={img.id} />
					))
				}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category : PropTypes.string.isRequired,
}