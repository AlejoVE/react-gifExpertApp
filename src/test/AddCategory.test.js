import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../components/AddCategory";

describe("Tests in component  <AddCategory />", () => {
  test("should throw an error if I dont pass  the setCategories property", () => {
    let wrapper = <AddCategory />;

    expect(wrapper).toThrowErrorMatchingSnapshot();
  });
});
