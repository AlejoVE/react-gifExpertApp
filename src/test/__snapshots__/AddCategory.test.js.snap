// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in component  <AddCategory /> should throw an error if I dont pass  the setCategories property 1`] = `"received is not a function"`;
